wm.c

```diff
@@ -1199,6 +1206,14 @@ is_maxed(struct client *client)
 		|| client->monocled;
 }
 
+static bool
+is_gridded(struct client *client)
```

Comment:
This would indeed be a nice feature.

I changed the naming of functions `is_maxed` to `is_special` and `unmaximize_window` to `reset_window` to reflect the fact that a window is now not only maximized but also possibly gridded.

I have one remark though:
By default, "special" (or previously "maximized") windows do not react to mouse commands (moving, resizing). Do you think it should still be the case with windows that are simply gridded? To me it would make sense to be able to move them freely with the mouse (and "reset" or "unmaximize" them in the process) since they do not necessarily cover the entire screen.

I left the default behaviour since this could be implemented in another PR